### PR TITLE
[fix](catalog) do not call makeSureInitialized when create/drop table/db from hms meta event

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

Supplement to https://github.com/apache/doris/pull/21104

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 50.87 seconds
 stream load tsv:          509 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17170834585 Bytes

File: fe/fe-core/src/main/java/org/apache/doris/catalog/external/HMSExternalDatabase.java [hunk 1]
```diff
@@ -74,7 +74,17 @@ public void dropTable(String tableName) {
     }
 
     @Override
-    public void replayCreateTableFromEvent(String tableName, long tableId) {
+    public void dropTableForReplay(String tableName) {
+        LOG.debug("replayDropTableFromEvent [{}]", tableName);
+        Long tableId = tableNameToId.remove(tableName);
+        if (tableId == null) {
+            LOG.warn("replayDropTableFromEvent [{}] failed", tableName);
+        }
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/external/HMSExternalDatabase.java:82]: missing `else`?

**Comment by morningman** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by zddr** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.78 seconds
 stream load tsv:          513 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         19 seconds loaded 2358488459 Bytes, about 118 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17161592610 Bytes

**Review by morningman — DISMISSED** [review]
LGTM

**Review by dutyu — APPROVED** [review]
LGTM

**Comment by zddr** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.99 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17161971208 Bytes

**Review by dutyu — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**